src/Hilo/Commands/HiloReturnCommand.ts

```diff
@@ -19,4 +23,16 @@ export class HiloReturnCommand extends RavenCommand {
     this.params = {tag: this.tag, last: this.last, end: this.end};
     this.endPoint = StringUtil.format('{url}/databases/{database}/hilo/return', serverNode);
   }
+
+  public setResponse(response: IResponse): IRavenResponse | IRavenResponse[] | void {
+    try {
+      super.setResponse(response);
+    } catch (exception) {
```

Comment:
This exception should be caught in line 111 of DocumentStore.ts - not here. See C#'s DocumentStore `Dispose()` method.

```
+  public async dispose(): Promise<IDocumentStore> {
      return this._generator.returnUnusedRange()
              .catch(() =>{})
              .then(...
```